fog/report/connection/src/lib.rs

```diff
@@ -1,172 +1,104 @@
+        uri: &FogUri,
+    ) -> Result<(), Error> {
+        match responses.entry(uri.to_string()) {
+            std::collections::btree_map::Entry::Occupied(_) => Ok(()),
+            std::collections::btree_map::Entry::Vacant(ent) => {
+                let logger = self.logger.new(o!("mc.fog.cxn" => uri.to_string()));
+
+                // Build channel to this URI
+                let ch = ChannelBuilder::default_channel_builder(self.env.clone())
+                    .connect_to_uri(uri, &logger);
+                let report_grpc_client = report_grpc::ReportApiClient::new(ch);
+
+                // Request reports
+                let req = mc_fog_api::report::ReportRequest::new();
+                let resp = report_grpc_client.get_reports(&req)?;
+
+                if resp.reports.len() == 0 {
+                    log::warn!(
+                        self.logger,
+                        "Report server at {} has no available reports",
+                        uri
+                    );
+                    return Err(Error::NoReports);
+                }
+
+                // Store entire response, for later validation against measurement and public addresses
+                ent.insert(resp);
+                Ok(())
+            }
+        }
+    }
+}
+
+/// Errors that can occur during GrpcFogReportConnection operation
 #[derive(Debug, Display)]
 pub enum Error {
-    /// Recipient doesn't have fog
-    RecipientHasNoFog,
-    /// Invalid fog url: {0}
-    InvalidFogUrl(UriParseError),
     /// grpc failure: {0}
     Rpc(grpcio::Error),
-    /// deserialization failed: {0}
-    DeserializationFailed(mc_util_serial::decode::Error),
-    /// report rejected: {0}
-    Rejected(IngestReportError),
     /// Fog Report Server has no available reports
     NoReports,
-    /// Matching report not found
-    NotFound,
-}
-
-impl From<UriParseError> for Error {
-    fn from(src: UriParseError) -> Self {
-        Self::InvalidFogUrl(src)
-    }
 }
 
 impl From<grpcio::Error> for Error {
     fn from(src: grpcio::Error) -> Self {
         Self::Rpc(src)
     }
 }
-
-impl From<mc_util_serial::decode::Error> for Error {
-    fn from(src: mc_util_serial::decode::Error) -> Self {
-        Self::DeserializationFailed(src)
-    }
-}
-
-impl From<IngestReportError> for Error {
-    fn from(src: IngestReportError) -> Self {
-        Self::Rejected(src)
-    }
-}
-
-/// Fog Pubkey resolver based on grpcio
-pub struct GrpcFogPubkeyResolver {
-    /// Ingest report verifier
-    verifier: IngestReportVerifier,
-    /// The logging instance
-    logger: Logger,
-    /// grpc environment
-    env: Arc<Environment>,
-}
-
-impl GrpcFogPubkeyResolver {
-    pub fn new(verifier: &Verifier, env: Arc<Environment>, logger: Logger) -> Self {
-        Self {
-            verifier: IngestReportVerifier::from(verifier),
-            env,
-            logger,
-        }
-    }
-}
-
-impl FogPubkeyResolver for GrpcFogPubkeyResolver {
```

Comment:
the grpc object no longer implements `FogPubkeyResolver` after this commit, it just produces `FogReportResponses`